Kernel/ACPI/Parser.h

```diff
@@ -9,22 +9,44 @@
 #include <AK/Types.h>
 #include <Kernel/ACPI/Definitions.h>
 #include <Kernel/ACPI/Initialize.h>
-#include <Kernel/FileSystem/File.h>
+#include <Kernel/API/SystemExposed.h>
 #include <Kernel/PhysicalAddress.h>
 #include <Kernel/VM/Region.h>
 #include <Kernel/VirtualAddress.h>
 
 namespace Kernel {
 namespace ACPI {
 
+class ExposedFolder : public SystemExposedFolder {
+public:
+    static void initialize();
+
+private:
+    ExposedFolder();
+};
+
+class ExposedComponent : public SystemExposedComponent {
+public:
+    static NonnullRefPtr<ExposedComponent> create(String name, PhysicalAddress, size_t table_size);
+
+    virtual KResultOr<size_t> read_bytes(off_t, size_t, UserOrKernelBuffer&, FileDescription*) const override;
+
+protected:
+    OwnPtr<KBuffer> try_to_generate_buffer() const;
+    ExposedComponent(String name, PhysicalAddress, size_t table_size);
+
+    PhysicalAddress m_paddr;
+    size_t m_length;
+};
+
 class Parser {
 public:
     static Parser* the();
 
     template<typename ParserType>
     static void initialize(PhysicalAddress rsdp)
     {
-        set_the(*new ParserType(rsdp));
+        set_the(*new (nothrow) ParserType(rsdp));
```

Comment:
This will cause a null pointer dereference if allocation fails.


`nothrow` new should only be used in conjunction with `adopt_ref_if_nonnull`, `adopt_own_if_nonnull`, or if you want a raw pointer that might be null. Sorry for the confusion about it. Will be better explained in #8295.

Reply:
Got it, thanks for the explanation :)